lib/oli/resources/collaboration.ex

```diff
@@ -130,10 +131,37 @@ defmodule Oli.Resources.Collaboration do
               "select max(inserted_at) from posts where section_id = ? and resource_id = ?",
               section.id,
               page_revision.resource_id
-            )
+            ),
+          count: over(count(page_revision.id))
         }
       )
-    )
+
+    query =
+      case opts[:limit] do
+        nil ->
+          query
+
+        limit ->
+          query
+          |> limit(^limit)
+      end
+
+    query =
+      case opts[:offset] do
+        nil ->
+          query
+
+        offset ->
+          query
+          |> offset(^offset)
+      end
```

Comment:
I think we can improve readability piping this (and probably reuse those functions in other queries), WDYT?:

```elixir
    from(...)
    |> maybe_add_query_limit(opts[:limit])
    |> maybe_add_query_offset(opts[:offset])
```

```elixir
  defp maybe_add_query_limit(query, nil), do: query
  defp maybe_add_query_limit(query, limit), do: limit(query, ^limit)

  defp maybe_add_query_offset(query, nil), do: query
  defp maybe_add_query_offset(query, offset), do: offset(query, ^offset)
```

Reply:
Actually, we could pipe all: 

```elixir
from(
      ...
    )
    |> maybe_add_query_limit(opts[:limit])
    |> maybe_add_query_offset(opts[:offset])
    |> Repo.all()
    |> case do
      [] ->
        {0, []}

      [first_post | _rest] = posts ->
        {first_post.count, Enum.map(posts, &Map.delete(&1, :count))}
    end
```